src/components/BSelect/BSelect.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+  <div>
+    <div>
+      <label>{{ selectLabel }} <span v-if=isRequired aria-label='Required'>*</span></label>
+    </div>
+    <div 
+      :aria-expanded="opened"
+      :aria-owns="id"
+      :aria-label="selectLabel"
+      :aria-required="isRequired"
+      role="combobox"
+      tabindex="0"
+      v-model="selectLabel"
+      :id="'dropdown' + id"
+      @keyup="keyUpHandler(selectedOption)"
+    >
+      <ul
+        :class="ulClass"
+        role="listbox"
+        :id="id"
+      >
+        <li 
+          :class="dropdownClass" 
+          @click="openList"
+          >{{ selectedOption ? selectedOption.text : placeholder }}
+          <span class="dropdown-arrow"><i class="fa fa-caret-down" aria-hidden="true"></i></span>
+        </li>
+        <li
+          v-if=opened
+          v-for="option in options"
+          :class="liClass"
+          :key="'selectOption' + option.id"
+          :id="'option' + option.id"
+          :value=option.id
+          :aria-selected="selectedOption && selectedOption.id === option.id"
+          @click="selectOption(option)"
```

Comment:
Minor thing - could we keep the handlers and directives grouped together? Something like

    <li
          v-if=opened
          v-for="option in options"
          :class="liClass"
          :key="'selectOption' + option.id"
          :id="'option' + option.id"
          :value=option.id
          :aria-selected="selectedOption && selectedOption.id === option.id"
          :ref="'option' + option.id"
          @keyup="keyUpHandler(option)"
          @click="selectOption(option)"
          role="option"
          tabindex='-1'
        > {{ option.text }}
        </li>

Reply:
Same for the other places where they're mixed.

Reply:
there's even a recommended order: https://vuejs.org/v2/style-guide/#Element-attribute-order-recommended